python/lsst/ip/isr/isrFunctions.py

```diff
@@ -1222,3 +1224,65 @@ def countMaskedPixels(maskedIm, maskPlane):
     maskBit = maskedIm.mask.getPlaneBitMask(maskPlane)
     nPix = numpy.where(numpy.bitwise_and(maskedIm.mask.array, maskBit))[0].flatten().size
     return nPix
+
+
+def getExposureGains(exposure):
+    """Get the per-amplifier gains used for this exposure.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        The exposure to find gains for.
+
+    Returns
+    -------
+    gains : `dict` [`str` `float`]
+        Dictionary of gain values, keyed by amplifier name.
+    """
+    det = exposure.getDetector()
+    if det is None:
+        return dict()
```

Comment:
In keeping this consistent with other parts of DM, it might be better to return None here or a dict of NaNs rather than an empty dict.

Reply:
Oh, interesting. I asked @erykoff about this and he suggested the empty dict.

Reply:
Eli confirmed using an empty dict.